### PR TITLE
Make sure search only happen for the desired index

### PR DESCRIPTION
**Pull request by purbon**
Make sure index is passed to search so we can control where search happen.

### CODE REVIEW

File: lib/logstash/filters/elasticsearch.rb [hunk 1]
```diff
@@ -84,7 +84,7 @@ def filter(event)
     begin
       query_str = event.sprintf(@query)
 
-      params = { :q => query_str, :size => result_size }
+      params = { :q => query_str, :size => result_size, :index => @index }
```
Comment (jsvd) [inline lib/logstash/filters/elasticsearch.rb]: does this mean that setting the index [here](https://github.com/purbon/logstash-filter-elasticsearch/blob/f9c3103c2ad91e5151c9cc0b0f01e0274fce667f/lib/logstash/filters/elasticsearch.rb#L78) does nothing?
if so we should remove it.

Also, is the default  `:index => ""` going to query all indices? Is it the same as `_all`?

Comment (purbon) [inline lib/logstash/filters/elasticsearch.rb]: 1.- It looks like is not being applied as expected as I initially thought, I will have to look closer to understand more. *I agree if does not work as expected should go*, removing this.

2.- yes, https://github.com/elastic/elasticsearch-ruby/blob/master/elasticsearch-api/lib/elasticsearch/api/actions/search.rb#L65 they are the same.

**Comment by purbon** [comment]
@jsvd thanks, I updated this PR and the backport with the clean up of the index param.

**Comment by jsvd** [comment]
LGTM

**Pull request closed without merging**